components/sup/src/manager/mod.rs

```diff
@@ -182,6 +182,7 @@ pub struct ManagerConfig {
     pub organization: Option<String>,
     pub watch_peer_file: Option<String>,
     pub tls_files: Option<(PathBuf, PathBuf)>,
+    pub no_listen_gossip: bool,
```

Comment:
Instead of adding this `bool`, why not make the `gossip_listen` field into an `Option`?

Reply:
Out of curiosity would that just be for the sake of standardization or is there a utilization difference between `bool` vs `Some()` or `None`?

Reply:
I think it's more to prevent having a bunch of struct fields than anything. If you're not going to be listening to gossip, it makes sense to fold that into the `gossip_listen` field as a `None` then having another field to keep track of. 

The more Rust I write, the more I think booleans are a kind of code smell that indicates that a more expressive solution is hiding somewhere.

(No, booleans aren't always a bad thing, but they do cause me to think a bit deeper.)

Reply:
In this case, it's the difference between having two fields:
```rust
   gossip_listen: GossipListenAddr,
   no_listen_gossip: bool,
```
and one:
```rust
   gossip_listen: Option<GossipListenAddr>,
```
The biggest reason to prefer the latter is that the former allows for the existence of an inconsistent state. Any time `no_listen_gossip` was `false`, there would still have to be an addr value in `gossip_listen`, but wouldn't make sense. If there were a value in there, code could access it and try to listen when we don't want it to. Combining them into one type prevents that situation by ensuring that in the case we don't want to listen, there is no addr value we could possibly use to do so. Code that tried would fail to compile and we'd catch the logic error early.

Reply:
What are your thoughts on just using the `--listen-gossip` option and allowing the user to pass `false`?

Reply:
I'm not a fan because that would imply that `--listen-gossip` accepts a `bool`, and it doesn't. If `--listen-gossip false` is valid, the fact that `--listen-gossip true` is non-sensical is surprising.

Is there something you don't like about `<--listen-gossip <LISTEN_GOSSIP>|--no-listen-gossip>`? We have other instances where an option and a flag are mutually exclusive.

Reply:
Eh, it wasn't a great suggestion. It's not that I don't like having the `--no-listen-gossip` flag, I'm just having difficulties getting it to work properly with clap.